library/java/src/io/envoyproxy/envoymobile/engine/JniBridgeUtility.java

```diff
@@ -0,0 +1,28 @@
+package io.envoyproxy.envoymobile.engine;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Class to assist with passing types from the JVM to native code. Currently supports
+ * HTTP headers.
+ */
+final class JniBridgeUtility {
+
+  private JniBridgeUtility() {}
+
+  public static byte[][] toJniHeaders(Map<String, List<String>> headers) {
```

Comment:
Could we add a quick test here?

Reply:
Sure (that was part of why I split it out to begin with). As it's existing logic that's been moved, I'll admit a slight preference for doing that as a follow-up though. wdyt?

Reply:
Sounds good. Let's do it as a follow up